batch_consumer.go

```diff
@@ -125,6 +128,11 @@ func chunkMessages(allMessages *[]*Message, chunkSize int) [][]*Message {
 func (b *batchConsumer) consume(allMessages *[]*Message, commitMessages *[]kafka.Message) {
 	chunks := chunkMessages(allMessages, b.messageGroupLimit)
 
+	if b.preBatchFn != nil {
```

Comment:
Just to let you know, I only changed chunks, not all messages. Because allMessages is used for commitMessages. If I mutate, I have some issues of offset/lag, etc.